modules/gghealthd/src/sd_bus.c

```diff
@@ -317,3 +319,64 @@ GglError get_lifecycle_state(
     err = get_component_result(bus, unit_path, state);
     return err;
 }
+
+void reset_restart_counters(sd_bus *bus, const char *qualified_name) {
+    sd_bus_error error = SD_BUS_ERROR_NULL;
+    sd_bus_message *reply = NULL;
+    GGL_LOGT("Issuing systemctl reset-failed for %s", qualified_name);
+    int ret = sd_bus_call_method(
+        bus,
+        "org.freedesktop.systemd1",
+        "/org/freedesktop/systemd1",
+        "org.freedesktop.systemd1.Manager",
+        "ResetFailedUnit",
+        &error,
+        &reply,
+        "s",
+        qualified_name
+    );
+    if (ret < 0) {
+        GGL_LOGW(
+            "Failed to reset failure counter for %s (errno=%d) (name=%s) "
+            "(message=%s)",
+            qualified_name,
+            -ret,
+            error.name,
+            error.message
+        );
+    }
+    GGL_CLEANUP(sd_bus_error_free, error);
+    GGL_CLEANUP(sd_bus_message_unrefp, reply);
```

Comment:
Should probably declare cleanups before the if statement, as is done in restart_component